src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederatedRange.java

```diff
@@ -94,12 +105,19 @@ public int compareTo(FederatedRange o) {
 			if ( _beginDims[i] > o._beginDims[i])
 				return 1;
 		}
+		if (_overlapNum < o._overlapNum)
+			return -1;
+		if (_overlapNum > o._overlapNum)
+			return 1;
```

Comment:
how is it relevant which index it is?
if you are trying to keep the overlaps in a certain order, it is really not a needed feature. 
But then again if it really is necessary to sort the federated ranges i guess this is an okay way.

if you change _overlapNum to be a boolean, you could sort on Object pointers instead in last case.

Reply:
See other comment about putting into map.